nibabel/testing/__init__.py

```diff
@@ -62,6 +65,22 @@ def assert_allclose_safely(a, b, match_nans=True, rtol=1e-5, atol=1e-8):
     assert_true(np.allclose(a, b, rtol=rtol, atol=atol))
 
 
+def check_iteration(iterable):
+    """ Checks that an object can be iterated through without errors. """
+    try:
+        for _ in iterable:
+            pass
+    except:
+        return False
+
+    return True
+
+
+def assert_arrays_equal(arrays1, arrays2):
+    for arr1, arr2 in zip_longest(arrays1, arrays2, fillvalue=None):
```

Comment:
I guess this one is for comparing sequences of arrays?  I guess too that the `fillvalue=None` is intended to make this fail when the sequences are different length?   If so, it would be easier to understand written as:

```
assert_equal(len(arrays1), len(arrays2))
for arr1, arr2 in zip(arrays1, arrays2):
    assert_array_equal(arr1, arr2)
```

That will also avoid hitting the numpy deprecation warning comparing arrays with None:

```
In [6]: None == np.arange(10)
/Users/mb312/Library/Python/2.7/bin/ipython:1: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  #!/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
Out[6]: False
```